src/client/src/app/services/navigation.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from "@angular/core";
+import { Location } from "@angular/common";
+import { Router, NavigationEnd } from "@angular/router";
+
+@Injectable({providedIn: 'root'})
+export class NavigationService{
+    private history: string[] = []
+
+    constructor(private router: Router, private location: Location) {
+        this.router.events.subscribe((event) => {
+            if (event instanceof NavigationEnd) {
```

Comment:
I tried to console log the history to see how the router events work I guess but I am still kind of lost on how this works lol. Just want to put a comment here so you will see and maybe explain to me what this does for my learning?

Reply:
Going to resolve this. Anyways, this is only a question that can be researched later on.